### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by nadjaheitmann**
I closed #5 in favor of this PR

@amirfefer The CI tests are currently failing because the resolver (#4) is missing. Unfortunately, I did not get the tests to work with current state of #4. If you want to try, you can checkout this PR and cherry-pick your commit from #4.

### CODE REVIEW

**Comment by nadjaheitmann** [comment]
Here is the CI run that ran for this PR: https://github.com/ATIX-AG/hammer-cli-foreman-puppet/runs/2903580045?check_suite_focus=true

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:17]: We run on Ruby 2.5 and 2.7 now. I'd also prefer the multi line approach since it's easier to maintain IMHO.
```suggestion
        ruby:
          - 2.5
          - 2.7
```

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:34]:
```suggestion
          repository: theforeman/hammer-cli-foreman
```

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:51]: If you call `setup-ruby` at this point (with `bundler-cache: true`) you actually get bundle install with cache.

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:45]: Debug code?

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:46]: Shouldn't you also need to set up hammer-cli here (since you cloned that repo as well)

File: hammer_cli_foreman_puppet.gemspec
Comment (ekohl) [inline hammer_cli_foreman_puppet.gemspec:21]:
```suggestion
  spec.add_dependency 'hammer_cli_foreman', '>= 3.0.0', '< 4.0.0'
```

File: Rakefile
Comment (ekohl) [inline Rakefile:12]:
```suggestion
task :default => :test
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [inline .github/workflows/unit_tests.yaml:41]: This shouldn't be needed now that you set up Ruby after cloning

**Review by ekohl — COMMENTED** [review]
The trick I found to get Github Actions to run new actions is to push the branch to upstream - you can delete it pretty much immediately after and then continue working on your fork.

File: .github/workflows/unit_tests.yaml
Comment (nadjaheitmann) [reply .github/workflows/unit_tests.yaml:41]: I get another dependency error if I remove it https://github.com/ATIX-AG/hammer-cli-foreman-puppet/runs/3201605023?check_suite_focus=true

@ekohl Thank you very much for your feedback!

**Comment by nadjaheitmann** [comment]
> The trick I found to get Github Actions to run new actions is to push the branch to upstream - you can delete it pretty much immediately after and then continue working on your fork.

I do not have enough permissions to push to upstream.

File: .github/workflows/unit_tests.yaml
Comment (nadjaheitmann) [reply .github/workflows/unit_tests.yaml:33]: Should be setup by bundle, so I removed it.

**Comment by ekohl** [comment]
> I do not have enough permissions to push to upstream.

I've done so now. It should now continue to run the workflows from unit_test.yml, even if you push a new version.

File: .github/workflows/unit_tests.yaml
Comment (ekohl) [reply .github/workflows/unit_tests.yaml:41]: I think you need to call `setup-ruby` with `working-directory: hammer-cli-foreman-puppet` (https://github.com/ruby/setup-ruby#working-directory).

File: test/data/README.md
Comment (ekohl) [inline test/data/README.md:25]:
```suggestion
    FOREMAN_VERSION = Gem::Version.new(ENV['TEST_API_VERSION'] || '3.0')
```

File: test/test_helper.rb
Comment (ekohl) [inline test/test_helper.rb:20]:
```suggestion
FOREMAN_VERSION = Gem::Version.new(ENV['TEST_API_VERSION'] || '3.0')
```

**Review by ekohl — COMMENTED** [review]
You added the foreman 2.5 API test data, but I think you should really only aim at Foreman 3.0.

**Comment by nadjaheitmann** [comment]
> You added the foreman 2.5 API test data, but I think you should really only aim at Foreman 3.0.

Let me see if I can fix that, I also need to create 3.0 API test data...

**Comment by ofedoren** [comment]
> You added the foreman 2.5 API test data, but I think you should really only aim at Foreman 3.0.

Just to mention: I agree that we need to use Foreman 3.0 with Foreman Puppet plugin enabled to generate the API docs for the tests.

**Comment by nadjaheitmann** [comment]
> > You added the foreman 2.5 API test data, but I think you should really only aim at Foreman 3.0.
> 
> Just to mention: I agree that we need to use Foreman 3.0 with Foreman Puppet plugin enabled to generate the API docs for the tests.

I used exactly this setup to generate test data on my personal Forklift. Foreman core https://github.com/theforeman/foreman/commit/fc082e8d753ead827e3ad0f16773008621f684b6 and foreman puppet https://github.com/theforeman/foreman_puppet/commit/3069d9022c41b69252932a5679d1735e28adb84f

File: hammer_cli_foreman_puppet.gemspec
Comment (ofedoren) [inline hammer_cli_foreman_puppet.gemspec:21]: I'd set it to `'>= 2.0.0', '< 4.0.0'`

**Review by ofedoren — COMMENTED** [review]
I've tested this PR with https://github.com/theforeman/hammer-cli-foreman-puppet/pull/4 and https://github.com/theforeman/hammer-cli-foreman-puppet/pull/16 and https://github.com/theforeman/hammer-cli-foreman-puppet/pull/18 being merged and locally all the tests pass.

I'd approve, but I still think that it's too early to pin for `>= 3.0.0` since there is no such version yet.

File: hammer_cli_foreman_puppet.gemspec
Comment (ekohl) [reply hammer_cli_foreman_puppet.gemspec:21]: Hasn't hammer-cli been bumped at least in nightly yet?

**Comment by ekohl** [comment]
> I'd approve, but I still think that it's too early to pin for `>= 3.0.0` since there is no such version yet.

But we've found that with hammer 2.5 this plugin just fails to load because it has overlapping definitions. Unless something changed in the past weeks.

**Comment by ofedoren** [comment]
> > I'd approve, but I still think that it's too early to pin for `>= 3.0.0` since there is no such version yet.
> 
> But we've found that with hammer 2.5 this plugin just fails to load because it has overlapping definitions. Unless something changed in the past weeks.

Yes, it will. Puppet extraction was merged, but there is no released version with it. So currently this plugin will work with current master branch only.

**Comment by ekohl** [comment]
But at least for RPMs we have nightly packages which should be 3.0.0. I'm not sure pretending to be compatible with 2.5 is correct.

**Comment by ofedoren** [comment]
> But at least for RPMs we have nightly packages which should be 3.0.0. I'm not sure pretending to be compatible with 2.5 is correct.

Agree, let's leave it to be `>= 3.0.0`.

**Comment by nadjaheitmann** [comment]
Do you want me to change anything else for this PR?

**Review by ekohl — APPROVED** [review]
:+1: from a CI perspective. I'll leave review of the actual changes to others.

**Review by ofedoren — APPROVED** [review]
ACK, I don't have merge permissions here :(

**Comment by ekohl** [comment]
I can help with that part.

**Pull request merged**
